ports/stm32f4/common-hal/busio/SPI.c

```diff
@@ -23,37 +24,320 @@
+    }
+    #endif
+    reset_pin_number(self->sck->pin->port,self->sck->pin->number);
+    reset_pin_number(self->mosi->pin->port,self->mosi->pin->number);
+    reset_pin_number(self->miso->pin->port,self->miso->pin->number);
+    self->sck = mp_const_none;
+    self->mosi = mp_const_none;
+    self->miso = mp_const_none;
+}
+
+static uint32_t stm32_baud_to_spi_div(uint32_t baudrate, uint16_t * prescaler) {
+    static const uint32_t baud_map[8][2] = {
+        {2,SPI_BAUDRATEPRESCALER_2},
+        {4,SPI_BAUDRATEPRESCALER_4},
+        {8,SPI_BAUDRATEPRESCALER_8},
+        {16,SPI_BAUDRATEPRESCALER_16},
+        {32,SPI_BAUDRATEPRESCALER_32},
+        {64,SPI_BAUDRATEPRESCALER_64},
+        {128,SPI_BAUDRATEPRESCALER_128},
+        {256,SPI_BAUDRATEPRESCALER_256}
+    };
+    size_t i = 0;
+    uint16_t divisor;
+    do {
+        divisor = baud_map[i][0];
+        if (baudrate >= (HAL_RCC_GetPCLK2Freq()/divisor)) {
+            *prescaler = divisor;
+            return baud_map[i][1];
+        }
+        i++;
+    } while (divisor != 256);
+    //should never get here
+    mp_raise_RuntimeError(translate("SPI Divisor error"));
+    return 0;
 }
 
 bool common_hal_busio_spi_configure(busio_spi_obj_t *self,
-         uint32_t baudrate, uint8_t polarity, uint8_t phase, uint8_t bits) {
-     return true;
+        uint32_t baudrate, uint8_t polarity, uint8_t phase, uint8_t bits) {
+    //This resets the SPI, so check before updating it redundantly
+    if (baudrate == self->baudrate && polarity== self->polarity 
+        && phase == self->phase && bits == self->bits) return true;
+
+    //Deinit SPI
+    HAL_SPI_DeInit(&self->handle);
+
+    if (bits == 8) self->handle.Init.DataSize = SPI_DATASIZE_8BIT;
+    else if (bits == 16) self->handle.Init.DataSize = SPI_DATASIZE_16BIT;
+    else return false;
+
+    if (polarity) self->handle.Init.CLKPolarity = SPI_POLARITY_HIGH;
+    else self->handle.Init.CLKPolarity = SPI_POLARITY_LOW;
+
+    if (phase) self->handle.Init.CLKPhase = SPI_PHASE_2EDGE;
+    else self->handle.Init.CLKPhase = SPI_PHASE_1EDGE;
+
+    self->handle.Init.BaudRatePrescaler = stm32_baud_to_spi_div(baudrate, &self->prescaler);
+
+    self->handle.Init.Mode = SPI_MODE_MASTER;
+    self->handle.Init.Direction = SPI_DIRECTION_2LINES;
+    self->handle.Init.NSS = SPI_NSS_SOFT;
+    self->handle.Init.FirstBit = SPI_FIRSTBIT_MSB;
+    self->handle.Init.TIMode = SPI_TIMODE_DISABLE;
+    self->handle.Init.CRCCalculation = SPI_CRCCALCULATION_DISABLE;
+    self->handle.Init.CRCPolynomial = 10;
+
+    if (HAL_SPI_Init(&self->handle) != HAL_OK)
+    {
+        mp_raise_RuntimeError(translate("SPI Re-init error"));
+    } else {
+        mp_printf(&mp_plat_print, "Configured! new spi clock speed:%u\n",(HAL_RCC_GetPCLK2Freq()/self->prescaler));
+        mp_printf(&mp_plat_print, "Prescaler is%u\n",self->prescaler);
+    }
+
+    self->baudrate = baudrate;
+    self->polarity = polarity;
+    self->phase = phase;
+    self->bits = bits;
```

Comment:
Save the prescaler too.

Reply:
The prescaler is actually modified by the stm32_baud_to_spi_div, which takes it in as a pointer reference. So it's also saved here with an updated value.